js/main.js

```diff
@@ -8,13 +8,24 @@ var entriesAnchor = document.querySelector('#entries-anchor');
 var newButton = document.querySelector('.new-button');
 var views = document.querySelectorAll('.view');
 var newEditEntry = document.querySelector('.new-edit-entry');
-var edit = false;
 
 switchViews(data.view);
 
 function saveEntry(event) {
   event.preventDefault();
-  if (edit === false) {
+  if (data.editing) {
+    data.editing.title = form.elements.title.value;
+    data.editing.photoUrl = form.elements['photo-url'].value;
+    data.editing.notes = form.elements.notes.value;
+    for (var x = 0; x < entries.children.length; x++) {
+      if (data.editing.id === parseInt(entries.children[x].getAttribute('data-entry-id'))) {
+        entries.children[x].querySelector('img').setAttribute('src', data.editing.photoUrl);
+        entries.children[x].querySelector('h2').textContent = data.editing.title;
+        entries.children[x].querySelector('p').textContent = data.editing.notes;
```

Comment:
Instead of this, I would just generate a new DOM tree with the updated values using your `loadEntry` function, and use the `replaceWith` method to replace the DOM tree you found with the edited entry's id.

https://developer.mozilla.org/en-US/docs/Web/API/Element/replaceWith